src/mlpack/methods/adaboost/adaboost_main.cpp

```diff
@@ -202,8 +202,10 @@ int main(int argc, char *argv[])
     else if (weakLearner == "perceptron")
       m.WeakLearnerType() = AdaBoostModel::WeakLearnerTypes::PERCEPTRON;
 
+    const size_t numClasses = arma::max(labels) + 1;
```

Comment:
Yes, that is required, but I am not sure how clear the documentation is.  I'll update it.